src/ansys/dpf/post/simulation.py

```diff
@@ -601,6 +601,26 @@ def _build_selection(
             return selection
         else:
             selection = Selection(server=self._model._server)
```

Comment:
This is the main change. Take it out of the if statements and takes into account the skin and external_layer flags passed to _build_selection.